fineract-loan/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/data/LoanSchedulePeriodData.java

```diff
@@ -265,6 +270,60 @@ private LoanSchedulePeriodData(final Integer periodNumber, final LocalDate fromD
         this.totalCredits = BigDecimal.ZERO;
     }
 
+    private LoanSchedulePeriodData(Integer periodNumber, LocalDate fromDate, LocalDate dueDate, BigDecimal principalDue,
```

Comment:
Eventually it would be nice to not have multiple constructor, but its just nice to have...

Reply:
I left a TODO and made a self note